src/r2mm/manager/SettingsDexieStore.ts

```diff
@@ -206,7 +206,7 @@ export interface ManagerSettingsInterfaceGlobal_V2 {
     favouriteGames: string[] | undefined;
     defaultGame: string | undefined;
     defaultStore: StorePlatform | undefined;
-    gameSelectionViewMode: string | undefined;
+    gameSelectionViewMode: GameSelectionViewMode;
```

Comment:
Shouldn't the new type also allow for `undefined`? I don't see anything in the code changes that guarantees this to never be undefined

Reply:
As far as I can tell there's no code changes needed for such guarantee. Again, *AFAICT*, currently the value can only be initiated by `SettingsDexieStore.createNewSettingsInstance()` and `.mapLegacyToV2()`. To answer your second comment, they both set the value to `GameSelectionViewMode.CARD` so it's already a default value. Once initiated, the value can only be set via `ManagerSettings.setGameSelectionViewMode()`, which doesn't accept `undefined` as a value.